packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+import { getClientName } from "./helpers/nameConstructors.js";
+import { NameType, camelCase, normalizeName } from "./helpers/nameUtils.js";
+import {
+  RLCModel,
+  RLCSampleGroup,
+  File as RLCFile,
+  Paths,
+  OperationMethod,
+  RLCSampleDetail,
+  SampleParameters,
+  SampleParameter,
+  PathMetadata,
+  OperationParameter,
+  ObjectSchema,
+  SchemaContext,
+  Schema,
+  ArraySchema,
+  DictionarySchema
+} from "./interfaces.js";
+import { sampleTemplate } from "./static/sampleTemplate.js";
+// @ts-ignore: to fix the handlebars issue
+import hbs from "handlebars";
+import * as path from "path";
+import { isObjectSchema } from "./helpers/schemaHelpers.js";
+import { getImmediateParentsNames } from "./buildObjectTypes.js";
+
+export function buildSamples(model: RLCModel) {
+  if (!model.options || !model.options.packageDetails) {
+    return;
+  }
+  const sampleGroups: RLCSampleGroup[] | undefined = model.sampleGroups;
+  if (!sampleGroups || sampleGroups.length === 0) {
+    return;
+  }
+  const sampleFiles: RLCFile[] = [];
+  for (const sampleGroup of sampleGroups) {
+    const sampleGroupFileContents = hbs.compile(sampleTemplate, {
+      noEscape: true
+    });
+    const filePath = path.join("samples-dev", `${sampleGroup.filename}.ts`);
+    sampleFiles.push({
+      path: filePath,
+      content: sampleGroupFileContents(sampleGroup)
+    });
+  }
+  return sampleFiles;
+}
+
+export function buildSamplesOnMockContent(model: RLCModel) {
+  if (model.options?.multiClient || model.options?.isModularLibrary) {
+    // Not support to generate if multiple clients
```

Comment:
According to our design decision, I think there will be no real multiple client case in RLC, I add this condition is more for backward compatibility because we have preview case in smoke test.